File: 080_remove_duplicates_sorted_array_ii.cc
class Solution {
public:
    int removeDuplicates(int A[], int n) {
        if(n == 0) return 0;
   		int i(0), j(1), prev = INT_MAX, count = 0;
		for(j = 0; j < n; ++j) {
		    if(A[j] == prev) count++;
		    else count = 1;
		    prev = A[j];
		    if(count <= 2) A[i++] = A[j]; 
		}
		return i;   
    }
};

File: _080_remove_duplicates_sorted_array_ii.java
/// Use an additional counter 

public class Solution {
    public int removeDuplicates(int[] nums) {
        //i: current element under investigation (jump over duplicates)
        //j: one past the end of processed array
        int n = nums.length;
        if (n <= 1) return n;
        int i = 0, j = 0;
        int count = 1;
        while (i < n) {
            count = 1;
            while (i < n-1 && nums[i+1] == nums[i]) { i++; count++;}
            nums[j++] = nums[i];
            if (count > 1) nums[j++] = nums[i];
            i++;
        }
        return j;
    }
}
